Mathlib/Combinatorics/Additive/FreimanHom.lean

```diff
@@ -0,0 +1,369 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.BigOperators.Multiset.Basic
+import Mathlib.Data.FunLike.Basic
+import Mathlib.Data.Set.Pointwise.Basic
+
+#align_import algebra.hom.freiman from "leanprover-community/mathlib"@"f694c7dead66f5d4c80f446c796a5aad14707f0e"
+
+/-!
+# Freiman homomorphisms
+
+In this file, we define Freiman homomorphisms and isomorphism.
+
+An `n`-Freiman homomorphism from `A` to `B` is a function `f : α → β` such that `f '' A ⊆ B` and
+`f x₁ * ... * f xₙ = f y₁ * ... * f yₙ` for all `x₁, ..., xₙ, y₁, ..., yₙ ∈ A` such that
+`x₁ * ... * xₙ = y₁ * ... * yₙ`. In particular, any `MulHom` is a Freiman homomorphism.
+
+An `n`-Freiman isomorphism from `A` to `B` is a function `f : α → β` bijective between `A` and `B`
+such that `f x₁ * ... * f xₙ = f y₁ * ... * f yₙ` for all `x₁, ..., xₙ, y₁, ..., yₙ ∈ A`
```

Comment:
There's an 'if and only if' missing somewhere here (the actual formal definition below is fine).